### PR TITLE
remove compatibility.h

### PR DESCRIPTION
**Pull request by Be-ing**
None of it is needed anymore.

### CODE REVIEW

**Comment by uklotzde** [comment]
I don't think so. We still need to support Qt 5.12.

**Comment by Be-ing** [comment]
Oh I did not realize QWidget::screen was added in Qt 5.14. I can add one conditional hack for that, but I think we can still get rid of compatibility.h.

**Comment by uklotzde** [comment]
Please don't remove the QUuid utility functions and tests, as these will still be needed in the future, e.g. for #2282.

**Comment by uklotzde** [comment]
I suggest to first merge #4308 and then continue with the remaining tasks.

**Comment by Be-ing** [comment]
I don't think we need to have unit tests for Qt. That's Qt's job.

**Comment by uklotzde** [comment]
> I don't think we need to have unit tests for Qt. That's Qt's job.

Qt doesn't provide a function that maps a null UUID to a null string.

**Comment by uklotzde** [comment]
I have removed the obsolete utility function and tests from #4308.

**Comment by Be-ing** [comment]
Rebased and added one special case hack for Qt < 5.14.

File: src/library/coverartdelegate.cpp [hunk 1]
```diff
@@ -126,8 +125,7 @@ void CoverArtDelegate::paintItem(
         VERIFY_OR_DEBUG_ASSERT(m_pCache) {
             return;
         }
-        const double scaleFactor =
-                getDevicePixelRatioF(qobject_cast<QWidget*>(parent()));
+        const double scaleFactor = qobject_cast<QWidget*>(parent())->devicePixelRatio();
```
Comment (uklotzde) [inline src/library/coverartdelegate.cpp:128]: Why `devicePixelRatio()` instead of `devicePixelRatioF()` as before?

Comment (Be-ing) [reply src/library/coverartdelegate.cpp:128]: an oversight, good catch

Comment (Be-ing) [reply src/library/coverartdelegate.cpp]: fixed

File: src/waveform/widgets/waveformwidgetabstract.cpp [hunk 2]
```diff
@@ -43,7 +42,7 @@ void WaveformWidgetAbstract::resize(int width, int height) {
     qreal devicePixelRatio = 1.0;
     if (m_widget) {
         m_widget->resize(width, height);
-        devicePixelRatio = getDevicePixelRatioF(m_widget);
+        devicePixelRatio = m_widget->devicePixelRatio();
```
Comment (uklotzde) [inline src/waveform/widgets/waveformwidgetabstract.cpp:45]: Please always check all occurrences, I only amend a single one!

**Review by uklotzde — CHANGES_REQUESTED** [review]
devicePixelRatio() -> devicePixelRatioF()

File: src/waveform/widgets/waveformwidgetabstract.cpp (hunk 2 above)
Comment (Be-ing) [reply src/waveform/widgets/waveformwidgetabstract.cpp]: Whoops, this got wrapped up in some other accidental changes I made which I then reverted so this one wasn't taken care of.

**Review by uklotzde — APPROVED** [review]

**Comment by Be-ing** [comment]
Merge?

**Comment by uklotzde** [comment]
LGTM

**Comment by Be-ing** [comment]
Thank you for the speedy review.

**Pull request merged**